recipes/librdkafka/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+import os
+
+from conans import CMake, ConanFile, tools
+
+
+class LibrdkafkaConan(ConanFile):
+    name = "librdkafka"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/edenhill/librdkafka"
+    description = "Librdkafka is an Apache Kafka C/C++ library designed with message delivery reliability and high performance in mind."
+
+    topics = ("kafka", "librdkafka")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "zlib": [True, False],
+        "zstd": [True, False],
+        "plugins": [True, False],
+        "ssl": [True, False],
+        "sasl": [True, False],
+        "lz4": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "zlib": False,
+        "zstd": False,
+        "plugins": False,
+        "ssl": False,
+        "sasl": False,
+        "lz4": False,
+    }
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+    sources_folder = "sources"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.ssl:
```

Comment:
```suggestion
        if self.options.ssl:
            self.output.warn("Forcing openssl:shared=True")
```

I don't like to force options, because you are creating a different package id from the expected profile. Of course, if you raise an error instead, it would break all CI jobs involving shared option, because we can't customize it internally for openssl only.
As we don't have any option, a warning should help to understand the intentional behavior.

Reply:
Sounds reasonable. And its forcing for both shared and static openssl.
```suggestion
        if self.options.ssl:
            self.output.warn(f"Forcing openssl:shared={self.options.shared}")
```

Reply:
But are you forcing because it's a project limitation? Is it REALLY required?

Reply:
No actually. I thought it would be a good idea not to mix static and shared libraries, especially with openssl. But there will be no problem with building the library without it.